src/AWS.Deploy.Common/Recommendation.cs

```diff
@@ -44,10 +45,30 @@ public Recommendation(RecipeDefinition recipe, ProjectDefinition projectDefiniti
             DeploymentBundle = new DeploymentBundle();
             DeploymentBundleSettings = deploymentBundleSettings;
 
+            GetRecommendationReplacementTokens(Recipe.OptionSettings);
+
             foreach (var replacement in additionalReplacements)
             {
-                if (!_replacementTokens.ContainsKey(replacement.Key))
-                    _replacementTokens[replacement.Key] = replacement.Value;
+                ReplacementTokens[replacement.Key] = replacement.Value;
+            }
+        }
+
+        private void GetRecommendationReplacementTokens(List<OptionSettingItem> optionSettings)

```

Comment:
Maybe rename to Collect or something more appropriate.

Reply:
Fixed